Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIDropdown/UIDropdownInstantiationSystem.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using DCL.Diagnostics;
+using DCL.ECSComponents;
+using DCL.Optimization.Pools;
+using DCL.SDKComponents.SceneUI.Classes;
+using DCL.SDKComponents.SceneUI.Components;
+using DCL.SDKComponents.SceneUI.Groups;
+using DCL.SDKComponents.SceneUI.Utils;
+using ECS.Abstract;
+
+namespace DCL.SDKComponents.SceneUI.Systems.UIDropdown
+{
+    [UpdateInGroup(typeof(SceneUIComponentInstantiationGroup))]
+    [LogCategory(ReportCategory.SCENE_UI)]
+    public partial class UIDropdownInstantiationSystem : BaseUnityLoopSystem
+    {
+        private readonly IComponentPool<DCLDropdown> dropdownsPool;
+
+        public UIDropdownInstantiationSystem(World world, IComponentPoolsRegistry poolsRegistry) : base(world)
+        {
+            dropdownsPool = poolsRegistry.GetReferenceTypePool<DCLDropdown>();
+        }
+
+        protected override void Update(float t)
+        {
+            InstantiateUIDropdownQuery(World);
+            UpdateUIDropdownQuery(World);
+        }
+
+        [Query]
+        [All(typeof(PBUiDropdown), typeof(UITransformComponent))]
+        [None(typeof(UIDropdownComponent))]
+        private void InstantiateUIDropdown(in Entity entity, ref UITransformComponent uiTransformComponent)
+        {
+            var dropdown = dropdownsPool.Get();
+            dropdown.Initialize($"UIDropdown (Entity {entity.Id})", "dcl-dropdown", "unity-base-popup-field__text");
```

Comment:
We can tolerate string creation but it's needed for debugging only in the editor. In the prod build it just creates an unneeded allocation: You could create a method with `#if UNITY_EDITOR` that would return a debug name. And under `else` just a constant name to avoid an allocation

Reply:
Good idea! I've done it for all the component's names 👍 https://github.com/decentraland/unity-explorer/pull/276/commits/aac3f3ff0c8933fc2349e2555ad9add214050b68